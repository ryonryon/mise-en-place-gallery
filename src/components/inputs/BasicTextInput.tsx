import styled from "@emotion/styled";
import * as React from "react";

import InputType from "./input-type";
import { secondary } from "../../constants/colors";

const StyledInput = styled.input`
  height: 24px;
  width: 70%;
  margin: 16px;
  padding: 4px;
  border: 1px solid ${secondary};
  color: ${secondary};
  border-radius: 4px;

  &::placeholder {
    color: ${secondary};
  }
`;

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  type: InputType;
}

const BasicTextInput = ({ type, ...rest }: Props) => {
  return <StyledInput type={type} {...rest} />;
};

BasicTextInput.defaultProps = {
  type: InputType.TEXT
};

export default BasicTextInput;
