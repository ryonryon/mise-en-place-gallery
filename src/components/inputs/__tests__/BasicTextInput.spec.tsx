import * as React from "react";
import * as renderer from "react-test-renderer";

import BasicTextInput from "../BasicTextInput";

describe("BasicTextInput test", () => {
  it("tests BasicTextInput renders correctly ", () => {
    const basicTextInput = renderer.create(<BasicTextInput />).toJSON();
    expect(basicTextInput).toMatchSnapshot();
  });
});
