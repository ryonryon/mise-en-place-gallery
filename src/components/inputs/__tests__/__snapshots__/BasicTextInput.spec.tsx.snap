// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicTextInput test tests BasicTextInput renders correctly  1`] = `
<input
  className="css-1puhmno"
  type="text"
/>
`;
